nexus/agents/api/views.py

```diff
@@ -418,3 +423,47 @@ def patch(self, request, project_uuid):
             "message": "Credentials updated successfully",
             "updated_credentials": updated_credentials
         })
+
+    def post(self, request, project_uuid):
+        credentials_data = request.data.get('credentials', {})
+        agent_uuid = request.data.get('agent_uuid')
+        is_confidential = request.data.get('is_confidential', True)
+
+        if not agent_uuid or not credentials_data:
+            return Response(
+                {"error": "agent_uuid and credentials are required"}, 
+                status=400
+            )
+
+        try:
+            agent = Agent.objects.get(uuid=agent_uuid)
+        except Agent.DoesNotExist:
+            return Response(
+                {"error": "Agent not found"}, 
+                status=404
+            )
+
+        created_credentials = []
+        for key, value in credentials_data.items():
+            existing_credential = Credential.objects.filter(
+                project_id=project_uuid,
+                key=key
+            ).first()
+
+            if not existing_credential:
+                # Create new credential if it doesn't exist
+              credential = Credential.objects.create(
+                  project_id=project_uuid,
```

Comment:
It is necessary to first check whether the user has permission on this project. The current format opens the door for users to create objects in projects that they do not have permission on.